File: 26.remove-duplicates-from-sorted-array.ts
/*
 * @lc app=leetcode id=26 lang=typescript
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
function removeDuplicates(nums: any[]): number {
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] === nums[i + 1]) {
      nums[i] = null;
      for (let j = i; j < nums.length; j++) {
        nums[j] = nums[j + 1];
      }
      nums.pop();
      i--;
    }
  }
  return nums.length;
}
// @lc code=end

File: 26.remove-duplicates-from-sorted-array.py
#
# @lc app=leetcode id=26 lang=python3
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        L = 1
        R = 1
        while R < len(nums):
            if nums[R] != nums[R-1]:
                nums[L] = nums[R]
                L += 1
            R += 1
        return L

# @lc code=end
